crates/vee/src/proof.rs

```diff
@@ -0,0 +1,158 @@
+/// generatesp proof for block based on its relation to the Merge and Capella upgrades
+/// in case of Ethereum BLocks. For Arbitrum, Optimism, it uses other methods to generate proofs
+use crate::protos::EthBlock;
+use alloy_primitives::B256;
+use ethportal_api::types::execution::header_with_proof::{
+    BlockHeaderProof,
+    // HistoricalRootsBlockProof, HistoricalSummariesBlockProof,
+    PreMergeAccumulatorProof,
+};
+
+/// The maximum number of block roots that can be stored in a [`BeaconState`]'s `block_roots` list.
+pub const SLOTS_PER_HISTORICAL_ROOT: usize = 8192;
+/// The merge block, inclusive, i.e., the block number below already counts to be post-merge.
+pub const MERGE_BLOCK: u64 = 15537394;
+/// The number of slots in an epoch.
+pub const SLOTS_PER_EPOCH: usize = 32;
+/// The number of slots in an era.
+pub const SLOTS_PER_ERA: usize = SLOTS_PER_HISTORICAL_ROOT;
+/// Slots are 0-indexed.
+/// See, for example, `https://beaconcha.in/slot/0`.
+pub const CAPELLA_START_EPOCH: usize = 194048;
+/// See [Upgrading Ethereum](https://eth2book.info/capella/part4/history/) for more information.
+/// The first slot number of the Deneb fork.
+pub const CAPELLA_START_SLOT: usize = CAPELLA_START_EPOCH * SLOTS_PER_EPOCH;
+/// The first era of the Deneb fork.
+pub const CAPELLA_START_ERA: usize =
+    (CAPELLA_START_EPOCH * SLOTS_PER_EPOCH) / SLOTS_PER_HISTORICAL_ROOT;
```

Comment:
I'm assuming the reason you're not reusing this from elsewhere in veemon is that this is just meant to show off your design, right?